Google.PowerShell/BigQuery/GcbqDataset.cs

```diff
@@ -0,0 +1,141 @@
+        /// characters or less. It must only contain letters (a-z, A-Z), numbers (0-9), or underscores (_).
+        /// </para>
+        /// </summary>
+        [Parameter(Position = 0, Mandatory = true, ParameterSetName = ParameterSetNames.ByValues)]
+        [ValidatePattern("[a-zA-Z0-9]")]
+        [ValidateLength(1, 1024)]
+        public string DatasetId { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The "friendly name" field for the dataset. Used for a descriptive name for the dataset.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByValues)]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The "description" field for the dataset. Used for a user-friendly description of the dataset.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByValues)]
+        public string Description { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The default lifetime for tables in the dataset to exist in ms. Minimum is 3600000 (1hr).  
+        /// When the expiration time is reached, the table will be automatically deleted.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByValues)]
+        [ValidateRange(3600000, long.MaxValue)]
+        public long Timeout { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            // Set up the Dataset based on parameters
+            Dataset newData;
+            switch (ParameterSetName)
+            {
+                case ParameterSetNames.ByObject:
+                    newData = Dataset;
+                    break;
+                case ParameterSetNames.ByValues:
+                    newData = new Dataset();
+                    newData.DatasetReference = new DatasetReference();
+                    newData.DatasetReference.DatasetId = DatasetId;
+                    newData.DatasetReference.ProjectId = Project;
+                    newData.FriendlyName = Name;
+                    newData.Description = Description;
+                    newData.DefaultTableExpirationMs = Timeout;
+                    break;
+                default:
+                    throw UnknownParameterSetException;
+            }
+
+            // Add the new dataset to the project supplied.
+            DatasetsResource.InsertRequest request = Service.Datasets.Insert(newData, Project);
+            Dataset response = request.Execute();
+            if (response != null)
+            {
+                WriteObject(response);
+            }
+            else
+            {
+                WriteError(new ErrorRecord(new Exception("400"), "Insert request to server failed.",
```

Comment:
Break this differently. The problem with the code as is, is that you are breaking after providing two parameters to the second type being instantiated. Read: this is harder to parse than it should be.

````
WriteError(new ErrorRecord(
    new Exception("400"),
    "Insert request to server failed.",
    ErrorCategory.InvalidArgument,
    newData));
````

If you are going to spill calls with mutliple args across a few lines, be sure to do it in such a way it is clear which function/constructor is being called. 

This isn't clear in the Java style guide, but generally try to follow that: https://engdoc.corp.google.com/eng/doc/devguide/java/styleguide.shtml?cl=head#s4.5.1-line-wrapping-where-to-break